File: python/8. KeyboardRow.py
'''
Given a List of words, return the words that can be typed using letters
of alphabet on only one row's of American keyboard
'''
class Solution(object):
    def findWords(self, words):
        out_words = []
        str1 = ['q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p']
        str2 = ['a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l']
        str3 = ['z', 'x', 'c', 'v', 'b', 'n', 'm']
        for word in words:
            res1, res2, res3 = 0, 0, 0
            for i in word:
                if i.lower() in str1:
                    res1 += 1
                if i.lower() in str2:
                    res2 += 1
                if i.lower() in str3:
                    res3 += 1
            if res1 == len(word) or res2 == len(word) or res3 == len(word):
                out_words += word.split(' ')
        return out_words

l = ["Hello", "Alaska", "Dad", "Peace"]
s = Solution()
print (s.findWords(l))

File: cpp/8. Keyboard Row.cpp
/*
Given a List of words, return the words that can be typed using letters
of alphabet on only one row's of American keyboard
*/

#include "stdafx.h" 
#include <iostream> 
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> findWords(vector<string>& words) {
		vector <string> out_words;
		string str1 = "qwertyuiop";
		string str2 = "asdfghjkl";
		string str3 = "zxcvbnm";

		for (int i = 0; i < words.size(); i++) {
			int res1 = 0, res2 = 0, res3 = 0;
			for (int j = 0; j < words[i].size(); j++) {
				char tmp = char(tolower(words[i][j]));
				if (str1.find(tmp) <= str1.size()) res1++;
				if (str2.find(tmp) <= str2.size()) res2++;
				if (str3.find(tmp) <= str3.size()) res3++;
			}
			if (res1 == words[i].size() || res2 == words[i].size() || res3 == words[i].size())
				out_words.push_back(words[i]);
		}
		return out_words;
    }
};

int main() { 
	vector <string> l;
	l.push_back("Hello");
	l.push_back("Alaska");
	l.push_back("Dad");
	l.push_back("Peace");

	Solution s;
	vector <string> res =  s.findWords(l);
	for (int i = 0; i < res.size(); i++) {
		cout << res.at(i) << ' ';
	}

	getchar();
	return 0; 
}